tests/test_Config.py

```diff
@@ -110,6 +120,17 @@ def testInit(self):
         self.assertEqual(self.comp.r.active.f, 3.0)
         self.assertEqual(self.comp.r["BBB"].f, 0.0)
 
+    def testDeprecationWarning(self):
+        """Test that a deprecated field emits a warning when it is set.
+        """
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
I think `with self.assertWarns(FutureWarning) as w:` is also an option here.

Reply:
Thanks, I don't know why I didn't use that, but it definitely works (had to modify the `assertIn`).